apps/src/templates/sectionProgress/sectionProgressRedux.js

```diff
@@ -222,14 +222,35 @@ export default function sectionProgress(state=initialState, action) {
           course.script_ids.forEach(id => idMap[id] = true);
         }
       });
-
       validScripts = validScripts.filter(script => idMap[script.id]);
+
+      var scriptId;
+      switch (true) {
+        // When there is a scriptId already in state.
+        case !!state.scriptId:
+          scriptId = state.scriptId;
+          break;
+        // When there is an assigned course, set scriptId to the first script in the assigned course.
```

Comment:
If there is an assigned script, it will already have been set in state.scriptId, because the setSection action is currently called before setValidScripts. It seems confusing to have this logic spread out between setSection and setValidScripts. 

I would recommend passing in the whole section here, then using both section.course_id and section.script.id to determine which section to assign.

If you don't have time to do this now, please add a comment indicating that (1) the logic is split between these two redux actions, and (2) it is required that setSection is called before setValidScripts in order for the right thing to happen in the case where a section has both a course and a script assigned.